userdocs/diagnostics/multiple-inheritance.md

```diff
@@ -0,0 +1,51 @@
+# Multiple Inheritance
+
+In some programming languages, a class can inherit the interface of multiple base classes. Known as multiple inheritance, this feature can add significant complexity to the language and is unsupported in Swift. Instead, Swift allows composition of interfaces using protocols.
+
+Consider the following example:
+
+```swift
+protocol Utensil { 
+    var name: String { get }
+} 
+
+/* ------------------------------ Serving Utensil ------------------------------ */
+protocol ServingUtensil: Utensil {
+    func serve()
+} 
+
+extension ServingUtensil {
+    func serve() { /* ... */ }
+}
+
+/* ------------------------------ Fork ------------------------------ */
+protocol Fork: Utensil {
+    func spear()
+}
+
+extension Fork {
+    func spear() { /* ... */ }
+}
+
+/* ------------------------------ Spoon ------------------------------ */
+protocol Spoon: Utensil {
+    func scoop()
+}
+
+extension Spoon {
+    func scoop() {/* ... */}
+}
+
+/* ------------------------------ Concrete Implementations ------------------------------ */
+struct CarvingFork: ServingUtensil, Fork {}
+
+struct Spork: Spoon, Fork {}
+
+struct Ladle: ServingUtensil, Spoon {}
```

Comment:
```suggestion
struct CarvingFork: ServingUtensil, Fork { /* ... */ }
struct Spork: Spoon, Fork { /* ... */ }
struct Ladle: ServingUtensil, Spoon { /* ... */ }
```

Since the example won't compile without concrete implementations, let's acknowledge that by inserting ellipses to indicate that we've elided parts of the code.